### PR TITLE
Use the standard GetConfigForClient in sshGRPCServer

### LINKED ISSUE #36964: Jump host fails with unknown certificate authority
Expected behavior:
`tsh ssh -J <leaf-proxy.example.com>` successfully connects to a leaf node.

Current behavior:
`tsh ssh` fails with error `unknown certificate authority`.

Bug details:
- Teleport version v15 alpha
- Recreation steps Log into leaf cluster via root, then run `tsh ssh -J <leaf-proxy.example.com> <leaf-node>`.
- Debug logs
```$ tsh ssh -J $HOSTNAME:4080 -d leaf-node
2024-01-19T15:00:38-08:00 INFO [CLIENT]    ALPN connection upgrade required for "andrew-ThinkPad-P14s-Gen-1:3080": false. client/api.go:709
2024-01-19T15:00:38-08:00 INFO [CLIENT]    no host login given. defaulting to andrew client/api.go:1045
2024-01-19T15:00:38-08:00 INFO [CLIENT]    [KEY AGENT] Connected to the system agent: "/tmp/ssh-XXXXXXP2Nmrj/agent.2370" client/api.go:4551
2024-01-19T15:00:38-08:00 DEBU [KEYSTORE]  Reading certificates from path "/home/andrew/.tsh/keys/andrew-ThinkPad-P14s-Gen-1/andrew-ssh/leaf-cert.pub". client/keystore.go:354
2024-01-19T15:00:38-08:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2024-01-20 10:55:43 +0000 UTC". client/client_store.go:106
2024-01-19T15:00:38-08:00 INFO [KEYAGENT]  Loading SSH key for user "andrew" and cluster "leaf". client/keyagent.go:195
2024-01-19T15:00:38-08:00 DEBU [CLIENT]    Overriding SSH proxy to JumpHosts's address "andrew-ThinkPad-P14s-Gen-1:4080" client/api.go:2877
2024-01-19T15:00:38-08:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2024-01-20 10:55:43 +0000 UTC". client/client_store.go:106
2024-01-19T15:00:38-08:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2024-01-20 10:55:43 +0000 UTC". client/client_store.go:106
2024-01-19T15:00:38-08:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2024-01-20 10:55:43 +0000 UTC". client/client_store.go:106

ERROR REPORT:
Original Error: trace.aggregate connection error: desc = &#34;error reading server preface: remote error: tls: unknown certificate authority&#34;
Stack Trace:
github.com/gravitational/teleport/api@v0.0.0/client/proxy/client.go:188 github.com/gravitational/teleport/api/client/proxy.NewClient
github.com/gravitational/teleport/lib/client/api.go:2820 github.com/gravitational/teleport/lib/client.(*TeleportClient).ConnectToCluster
github.com/gravitational/teleport/lib/client/api.go:1529 github.com/gravitational/teleport/lib/client.(*TeleportClient).SSH
github.com/gravitational/teleport/tool/tsh/common/tsh.go:3228 github.com/gravitational/teleport/tool/tsh/common.onSSH.func1.1
github.com/gravitational/teleport/lib/client/api.go:556 github.com/gravitational/teleport/lib/client.RetryWithRelogin
github.com/gravitational/teleport/tool/tsh/common/tsh.go:3227 github.com/gravitational/teleport/tool/tsh/common.onSSH.func1
github.com/gravitational/teleport/tool/tsh/common/tsh.go:3136 github.com/gravitational/teleport/tool/tsh/common.retryWithAccessRequest
github.com/gravitational/teleport/tool/tsh/common/tsh.go:3226 github.com/gravitational/teleport/tool/tsh/common.onSSH
github.com/gravitational/teleport/tool/tsh/common/tsh.go:1243 github.com/gravitational/teleport/tool/tsh/common.Run
github.com/gravitational/teleport/tool/tsh/common/tsh.go:541 github.com/gravitational/teleport/tool/tsh/common.Main
github.com/gravitational/teleport/tool/tsh/main.go:24 main.main
runtime/proc.go:267 runtime.main
runtime/asm_amd64.s:1650 runtime.goexit
User Message: connection error: desc = &#34;error reading server preface: remote error: tls: unknown certificate authority&#34;
```

Appears to be related to #30043, as jump hosts work in the commit before it.

### PR DESCRIPTION
**Pull request by espadolini**
The "SSH gRPC" service currently only trusts the host and user CAs for the local cluster; it should, however, since it's necessary for the tsh ProxyJump functionality (`tsh ssh -J addr-of-leaf-proxy user@host ...`). This PR changes the TLS config for the service to use the same `GetConfigForClient` used by most other servers, which will load user CAs on demand.

The changes to the test make it fail without the fix (in the "ssh jump host access" subtest) and succeed with; however, this PR introduces some minor go:debug hackery (in test code) to extend the system cert pool.
Fixes #36964

changelog: Fixed incompatibility between leaf clusters and ProxyJump

### CODE REVIEW

File: integration/helpers/x509.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import (
+	"crypto/x509"
+	"sync"
+	"testing"
+	_ "unsafe"
+)
+
+//go:linkname x509_systemRootsMu crypto/x509.systemRootsMu
+//go:linkname x509_systemRoots crypto/x509.systemRoots
```
Comment (zmb3) [inline integration/helpers/x509.go:27]: Can we at least do this in a _test.go file to be sure it doesn't ever affect a production build?

**Review by zmb3 — COMMENTED** [review]
I'm on the fence about the testing changes. They seem both complicated and brittle.

In this case I would be okay with merging without the tests since you've demonstrated already that the fix is good.

File: integration/helpers/x509.go (hunk 1 above)
Comment (espadolini) [reply integration/helpers/x509.go:27]: The issue with that is that other packages' test files don't exist when building tests.

Should I move this back to `tools/tsh/common/proxy_test.go` for now?

Comment (espadolini) [reply integration/helpers/x509.go]: I figured out that with `//go:debug x509usefallbackroots=1` (which is far more supported) we get to replace the system cert pool without using linkname hacks or changing all the test running environments to add the matching GODEBUG envvar (b986a46)

**Review by zmb3 — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**